### PR TITLE
Updated Containers with Vagrant section and added routing. Added new .sh script for building index.rst files.

### PR DESCRIPTION
**Pull request by a-olechtchouk**
Added further detail for creating multiple branches in the section  "Pushing your changes to the VPP Docs Repository". In that section I also fixed the subtitles so they would appear in the navigation drop-down on the left.

Created a new section "VPP with two Containers using Vagrant". Currently it just has installation information, but will have container support very soon.

Read the Docs link is below:
[https://a-olechtchoukvpp-docs.readthedocs.io/en/latest/reference/vppcontainers/twocontainers.html](https://a-olechtchoukvpp-docs.readthedocs.io/en/latest/reference/vppcontainers/twocontainers.html)

### CODE REVIEW

File: docs/gettingstarted/writingdocs/pushingapatch.rst [hunk 1]
```diff
@@ -103,24 +104,23 @@ To do so, fetch any changes that the main repo has made, and then merge them int
    $ git merge upstream/master
 
 
-Create a Branch
----------------
+.. note:: **This is optional, so don't do these commands if you just want one local branch!!!**
 
-At this point you may want to work on a branch. To create a branch create and checkout the branch.
+    You may want to have multiple branches, where each branch has its own different features, allowing you to have multiple pull requests out at a time. To create a new local branch:
```
Comment (mdr78) [inline docs/gettingstarted/writingdocs/pushingapatch.rst]: Change want to 'wish' or 'like'

File: docs/reference/vppcontainers/twocontainers.rst [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+Installing Vagrant
+__________________
+
+
+Now its time to install Vagrant.
+
+Here we are on a 64-bit version of CentOS, downloading and installing Vagrant 2.1.1:
+
+.. code-block:: shell
```
Comment (mdr78) [inline docs/reference/vppcontainers/twocontainers.rst]: Can we refer people to the hashicorp downloads page.

https://www.vagrantup.com/downloads.html

Referencing a specific version is going to get out of date very fast here.

File: docs/reference/vppcontainers/twocontainers.rst [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+This will create a *Vagrantfile* in your current directory, which you can further configure based on the requirements you want your VM to have. To learn more about configuration options, visit the `Vagrant Vagrantfile page <https://www.vagrantup.com/docs/vagrantfile/>`_.
+
+Alternatively, if you want to start with an empty Vagrantfile, you can create an empty text file that has this configuration option:
+
+.. code-block:: console
+
+   Vagrant.configure("2") do |config|
+     config.vm.box = "ubuntu/trusty64"
+   end
+
```
Comment (mdr78) [inline docs/reference/vppcontainers/twocontainers.rst]: stray carriage returns.

File: docs/reference/vppcontainers/twocontainers.rst [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+If you're on CentOS, follow the `steps here <https://wiki.centos.org/HowTos/Virtualization/VirtualBox#head-81de410879b8e7f18a127f638160e036ab99684e>`_.
+
+
+If you're on Ubuntu, perform:
+
+.. code-block:: shell
+
+   $ sudo apt-get install virtualbox 
+
+
```
Comment (mdr78) [inline docs/reference/vppcontainers/twocontainers.rst]: stray carriage returns.

File: docs/reference/vppcontainers/twocontainers.rst [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+In this section, we'll use Vagrant to run our VirtualBox VM. **Vagrant** automates the configuration of virtual environments by giving you the ability to create and destroy VM's quick and seemlessly.
+
+Installing VirtualBox
+_____________________
+
+First, download VirtualBox, which is virtualization software for creating VM's.
+
+If you're on CentOS, follow the `steps here <https://wiki.centos.org/HowTos/Virtualization/VirtualBox#head-81de410879b8e7f18a127f638160e036ab99684e>`_.
```
Comment (mdr78) [inline docs/reference/vppcontainers/twocontainers.rst]: If you are using CentOS, follow the `step here <https://wiki.centos.org/HowTos/Virtualization/VirtualBox>`_

File: docs/reference/vppcontainers/twocontainers.rst [hunk 6]
```diff
@@ -0,0 +1,117 @@
+.. _twocontainers:
+
+.. toctree::
+
+
+=====================================
+VPP with two Containers using Vagrant
+=====================================
+
```
Comment (mdr78) [inline docs/reference/vppcontainers/twocontainers.rst]: stray carriage return

File: docs/reference/vppcontainers/twocontainers.rst [hunk 7]
```diff
@@ -0,0 +1,117 @@
+.. _twocontainers:
+
+.. toctree::
+
```
Comment (mdr78) [inline docs/reference/vppcontainers/twocontainers.rst]: stray carriage return

File: docs/reference/vppcontainers/twocontainers.rst [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+=====================================
+VPP with two Containers using Vagrant
+=====================================
+
+
+Overview
+________
+
+This section will cover how to install a VM (Virtual Machine) for Vagrant, and how to use containers inside that VM.
```
Comment (mdr78) [inline docs/reference/vppcontainers/twocontainers.rst]: This section will describe how to install a Virtual Machine (VM) for Vagrant, and install containers inside that VM.

File: docs/reference/vppcontainers/twocontainers.rst [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+
+Containers are environments similar to VM's, but are known to be faster since they do not simulate  seperate kernels and hardware, as VM's do. You can read more about `Linux containers here <https://linuxcontainers.org/>`_.
+
+
```
Comment (mdr78) [inline docs/reference/vppcontainers/twocontainers.rst]: Stray carriage returns.

File: docs/reference/vppcontainers/twocontainers.rst [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+The syntax of Vagrantfiles is the programming language *Ruby*, but experience with Ruby is not completely necessary as most modifications to your  Vagrantfile is changing variable values. 
+
+The Vagrantfile creates a **Vagrant Box**, which is a "development-ready box" that can be copied to other machines to recreate the same environment. The `Vagrant website for boxes <https://app.vagrantup.com/boxes/search>`_  shows you all the available Vagrant Boxes containing different operating systems.
+
+
+Starting up
+^^^^^^^^^^^
+
+Begin by changing directories to where you want to install your box/VM.
```
Comment (mdr78) [inline docs/reference/vppcontainers/twocontainers.rst]: your Vagrant Box

**Comment by a-olechtchouk** [comment]
Added additional information for using Linux Containers. Yet figure out how to change Linux network interface names and connect the containers together.

New docs here: https://a-olechtchoukvpp-docs.readthedocs.io/en/latest/reference/vppcontainers/twocontainers.html

**Comment by zxiiro** [comment]
I'm not sure why buildHTML.sh is needed at all. The usual route to build Sphinx docs in most projects is to do it via tox.ini to manage the virtualenv for us.

In ODL we define the following tox.ini https://github.com/opendaylight/docs/blob/master/tox.ini#L19-L22 the "testenv:docs" section is important.

tox works in Windows / Mac / Linux so should be pretty cross platform compatible.

In versions I believe 1.7.x of Sphinx it even prints the location of the outputted html so that the user will know where to open the generated html.

I don't believe "open index.html" is a cross compatible way to open a file though at least on my Linux box I seem to be missing the "open" command.

Additionally if we hook this into global-jjb's RTD verify jobs in Jenkins you can get Jenkins to build and publish the generated docs to the log server we can allow reviewers to see the generated output of a change without having to build it themselves. See:

https://git.opendaylight.org/gerrit/73457

Look for the "Logs:" output for the rtd-verify job for example:

https://logs.opendaylight.org/releng/vex-yul-odl-jenkins-1/docs-rtd-verify-fluorine/763

On that logs page clicking "html" will show the generated output of the verify job for a specific patch.

**Pull request merged**